fedora/client/openidcclient.py

```diff
@@ -257,67 +240,64 @@ def _get_new_token(self, scopes):
         the valid cache, and then return the UUID.
         If the user cancelled (or we got another error), we will return None.
         """
-        def token_app(environ, start_response):
+        def _token_app(environ, start_response):
             query = environ['QUERY_STRING']
             split = query.split('&')
             kv = dict([v.split('=', 1) for v in split])
 
             if 'error' in kv:
-                self.debug('Error code returned: %s (%s)'
-                           % (kv['error'], kv.get('error_description')))
-                self.__retrieved_code = False
+                self.debug('Error code returned: %s (%s)',
+                           kv['error'], kv.get('error_description'))
+                self._retrieved_code = False
             else:
-                self.__retrieved_code = kv['code']
+                self._retrieved_code = kv['code']
 
             # Just return a message
             out = StringIO()
             out.write('You can close this window and return to the CLI')
             start_response('200 OK', [('Content-Type', 'text/plain')])
             return [out.getvalue()]
 
-        self.__retrieved_code = None
-        server = self._get_server(token_app)
+        self._retrieved_code = None
+        server = self._get_server(_token_app)
         if not server:
             raise Exception('We were unable to instantiate a webserver')
         return_uri = 'http://localhost:%i/' % server.socket.getsockname()[1]
-        query = {}
-        # We always silently inject scope openid so we get an id_token
-        if len(scopes) > 0:
-            query['scope'] = 'openid %s' % ' '.join(scopes)
-        else:
-            query['scope'] = 'openid'
-        query['response_type'] = 'code'
-        query['client_id'] = self.client_id
-        query['redirect_uri'] = return_uri
-        query['response_mode'] = 'query'
-        query = urlencode(query)
+        rquery = {}
+        rquery['scope'] = ' '.join(scopes)
+        rquery['response_type'] = 'code'
+        rquery['client_id'] = self.client_id
+        rquery['redirect_uri'] = return_uri
+        rquery['response_mode'] = 'query'
+        query = urlencode(rquery)
         authz_url = '%s?%s' % (self._idp_url('Authorization'), query)
         print('Please visit %s to grant authorization' % authz_url)
         webbrowser.open(authz_url)
         server.handle_request()
         server.server_close()
 
-        assert self.__retrieved_code is not None
-        if self.__retrieved_code is False:
+        assert self._retrieved_code is not None
+        if self._retrieved_code is False:
             # The user cancelled the request
-            self.__retrieved_code = None
+            self._retrieved_code = None
             self.debug('User cancelled')
             return None
 
         self.debug('We got an authorization code!')
-        resp = requests.post(
+        resp = requests.request(
```

Comment:
I could, but then I would need to mock out two paths, and have different code to dealing with mocking the responses.